File: exercicios/python/Exercício 03 - Implementação de Classes e Objetos (Pacote de Viagem) (25_08)/pacote_viagem.py
class PacoteViagem:
    def __init__(self, origem: str, destino: str, duracao: int, custo_unitario: float):
        self.__origem = origem
        self.__duracao = duracao
        self.__custo_unitario = custo_unitario
        self.__destino = destino

    @property
    def destino(self):
        return self.__destino

    @destino.setter
    def destino(self, destino: str):
        self.__destino = destino

    @property
    def duracao(self):
        return self.__duracao

    @duracao.setter
    def duracao(self, duracao: int):
        self.__duracao = duracao

    @property
    def custo_unitario(self):
        return self.__custo_unitario

    @custo_unitario.setter
    def custo_unitario(self, custo_unitario: float):
        self.__custo_unitario = custo_unitario

    @property
    def origem(self):
        return self.__origem

    @origem.setter
    def origem(self, origem: str):
        self.__origem = origem

File: exercicios/java/Exercicio03/src/main/java/org/example/PacoteViagem.java
package org.example;

public class PacoteViagem {
    private String origem;
    private String destino;
    private int duracao;
    private static double custoUnitario;

    public PacoteViagem(String origem, String destino, int duracao, double custoUnitario) {
        this.origem = origem;
        this.destino = destino;
        this.duracao = duracao;
        this.custoUnitario = custoUnitario;
    }
    public String getOrigem() {
        return origem;
    }
    public void setOrigem(String origem) {
        this.origem = origem;
    }
    public String getDestino() {
        return destino;
    }
    public void setDestino(String destino) {
        this.destino = destino;
    }
    public int getDuracao() {
        return duracao;
    }
    public void setDuracao(int duracao) {
        this.duracao = duracao;
    }
    public static double getCustoUnitario() {
        return custoUnitario;
    }
    public void setCustoUnitario(double custoUnitario) {
        this.custoUnitario = custoUnitario;
    }
}
